client/rest-high-level/src/main/java/org/elasticsearch/client/indices/PutIndexTemplateRequest.java

```diff
@@ -76,6 +75,8 @@
 
     private Integer version;
 
+    private TimeValue masterNodeTimeout = TimeValue.timeValueSeconds(30);
```

Comment:
you could use https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/TimedRequest.java here